src/Domain/Model/MembershipApplication.php

```diff
@@ -120,11 +121,11 @@ private function isCancellable(): bool {
 			return false;
 		}
 
-		if ( $this->payment->getPaymentMethod()->hasExternalProvider() ) {
+		if ( !$this->payment instanceof CancellablePayment ) {
```

Comment:
Please remove this. Membership is not supposed to know about payments, the cancel membership use case should  check with payment instead. We'll deal with the failing tests later.